edgedb/credentials.py

```diff
@@ -19,6 +22,10 @@ class Credentials(RequiredCredentials, total=False):
     database: typing.Optional[str]
 
 
+def get_credentials_path(instance_name: str) -> pathlib.Path:
+    return platform.config_dir() / "credentials" / (instance_name + ".json")
```

Comment:
I'd put an existence check here and fallback to `~/.edgedb` to preserve compatibility with unmigrated layout.  Otherwise a binding update done before `edgedb self migrate` will break user apps.

Reply:
Added in `config_dir()` so that `edgedb.toml` project link could also be compatible.